protocol/crosschain-erc20.md

```diff
@@ -0,0 +1,277 @@
+- Users can always unwrap their tokens to return to the original `ERC20`
+
+## **C. Deployed, non-upgradable `xERC20`**
+
+For tokens already deployed as non-upgradable `xERC20`, upgrading to implement ERC-7802 is not an option. To integrate with the Superchain while maintaining existing bridge infrastructure, an `ERC7802Adapter` can be used. This contract implements `ERC-7802`’s crosschainBurn and crosschainMint functions, translating them into the corresponding `xERC20` mint/burn operations.
+
+Since bridges will interact with the adapter instead of the token contract, the adapter must be deployed deterministically to ensure the same address across all chains.
+
+#### Raw Example:
+
+```solidity
+function crosschainMint(address _to, uint256 _amount) onlyBridge external {
+    XERC20.mint(_to, _amount);
+
+    emit CrosschainMint(_to, _amount, msg.sender);
+}
+```
+
+#### Properties:
+
+- Uses `SuperchainTokenBridge` or any other bridge that implements `ERC-7802`.
+- Token owner has to deploy one adapter for each chain using the same address.
+
+#### Setup Diagram:
+
+```mermaid
+sequenceDiagram
+  actor A0 as Owner
+  box L2 Chain
+	  participant P1 as Factory
+    participant P2 as Adapter
+    participant P3 as XERC20
+  end
+
+	A0 ->> P1: deployAdapter()
+	P1 ->> P2: constructor(XERC20)
+  A0 ->> P3: setLimit(Adapter...)
+
+```
+
+#### Usage Diagrams:
+
+```mermaid
+sequenceDiagram
+
+  actor A1 as Alice
+  box L2 Chain A
+    participant P1 as SuperchainTokenBridge
+    participant P2 as Adapter
+    participant P3 as XERC20 A
+  end
+
+  A1 ->> P1: sendERC20(Adapter...)
+  P1 ->> P2: crosschainBurn()
+  P2 ->> P3: burn()
+  P1 -->> P1: SendERC20()
+
+```
+
+```mermaid
+sequenceDiagram
+
+  box L2 Chain B
+    participant P5 as XERC20 B
+    participant P6 as Adapter'
+    participant P7 as SuperchainTokenBridge
+  end
+  actor A2 as Relayer
+
+  A2 ->> P7: relayERC20(Adapter'...)
+  P7 ->> P6: crosschainMint()
+  P6 ->> P5: mint()
+  P7 -->> P7: RelayERC20()
+```
+
+## CrosschainERC20Factory
+
+The Factory contract serves as a central deployment mechanism for all the components in our system. It provides methods to:
+
+1. Deploy new `CrosschainERC20` tokens and grant mint/burn permissions to the `SuperchainTokenBridge` or any other bridge that implements `ERC-7802`.
+2. Deploy Lockboxes for existing tokens that support `ERC20` interfaces
+3. Deploy Adapters for existing `xERC20` tokens
+
+**Key Features:**
+
+- Deterministic deployments using CREATE3
+- Registry of deployed contracts
+- Permission management for deployments
+- Single entry point for all deployments
+
+---
+
+## Alternatives Considered for Scenario C
+
+_Worth mentioning that the Lockbox solution described in section B is also compatible with this scenario, providing an alternative option if the token issuer prefers to wrap their `xERC20` into a new `CrosschainERC20` token_
+
+### **CrosschainERC20Converter**
```

Comment:
Not sure I understand this one: so the CrosschainERC20Converter contract is a CrosschainERC20 contract? What does the `undo` call do?